frontend/src/pages/profile/Profile.tsx

```
@@ -0,0 +1,570 @@
+      if (url) {
+        await axios.post(url);
+
+        // Optimistically update the UI *immediately*
+        if (isFollowing) {
+          setIsFollowing(false); // Just unfollowed
+        } else if (user.isPrivate && !followRequested) {
+          setFollowRequested(true); // Just sent a request
+        } else if (!user.isPrivate) {
+          setIsFollowing(true); // Just followed directly
+
+          if (user.uid != user.uid) {
```

Biome (1.9.4)

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)